File: phalcon/app/modules/admin/model/SysMenuAction.php
<?php

namespace app\modules\admin\model;

use app\model\BaseModel;

class SysMenuAction extends BaseModel{

  public $id;
  private $name;
  private $action;
  private $perm;

  /* 数据表 */
  public function getSource(){
    return "sys_menus_action";
  }
  
  /* 名称 */
  public function setName($name){
    $num = mb_strlen($name,"utf-8");
    if($num<2 || $num>6){
      return self::error('名称为2~6位字符！');
    }
    $this->name = $name;
  }
  public function getName(){
    return $this->name;
  }

  /* 命名 */
  public function setAction($action){
    if(!preg_match('/^[a-zA-Z]{3,16}$/',$action)){
      return self::error('命名为3~16位英文！');
    }
    $this->action = $action;
  }
  public function getAction(){
    return $this->action;
  }

  /* 权限 */
  public function setPerm($perm){
    if((int)$perm<1 || (int)$perm%2!=0){
      return self::error('权限为2的n次方！');
    }
    $this->perm = $perm;
  }
  public function getPerm(){
    return $this->perm;
  }

}

File: java/src/main/java/vip/webmis/java/model/SysMenuAction.java
package vip.webmis.java.model;

import vip.webmis.java.library.Safety;

/* 系统菜单表 */
public class SysMenuAction extends Model {

  public int id = 0;
  public String name = "";
  public String action = "";
  public int perm = 0;
  public String ico = "";
  
  /* 构造函数 */
  public SysMenuAction() {
    this.setSource("sys_menus_action"); //数据表
  }

  /* 名称 */
  public void setName(String val) throws Exception {
    int num = val.length();
    if(num<2 || num>6){
      error("名称为2~6位字符!");
    }
    name = val;
  }
  public String getName(){
    return name;
  }

  /* 命名 */
  public void setAction(String val) throws Exception {
    if(!Safety.test("^[a-zA-Z]{2,16}$",val)){
      error("命名为2~16位英文!");
    }
    action = val;
  }
  public String getAction(){
    return action;
  }

  /* 权限 */
  public void setPerm(Integer val) throws Exception {
    if(val<2 || val%2!=0){
      error("权限为2的n次方!");
    }
    perm = val;
  }
  public int getPerm(){
    return perm;
  }

}
